### ISSUE TITLE
Bzlmod support

### ISSUE THREAD

**Issue by mering** [tensorflow/tensorflow#62598]
### Issue type

Build/Install

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

Latest

### Custom code

No

### OS platform and distribution

Linux

### Bazel version

6

### Current behavior?

Please support bzlmod by adding a `MODULE.bazel` file . See the [migration guide](https://bazel.build/external/migration).

### Standalone code to reproduce the issue

```shell
This file currently doesn't exist.
```

**Comment by tilakrayal** [comment]
@mering,
Thank you for the request. Could you please elaborate about the Feature. Also, please specify the Use Cases for this feature. Thank you!

**Comment by mering** [comment]
@tilakrayal The current Bazel `WORKSPACE` approach has several flaws which are fixed by the migration to Bazel modules (e.g. not requiring dependants to specify *all* transitive dependencies). I can't describe it better with my own words than the [Bzlmod user guide](https://docs.bazel.build/versions/5.4.1/bzlmod.html).

**Comment by mihaimaruseac** [comment]
The team is working on adding bzlmod support, but since this is a big project this will take significant time

**Comment by lucien-perouze** [comment]
Do you have any info on the status of this feature ? Would love to have Bzlmod support to import TensorFlow seamlessly.

**Comment by lucien-perouze** [comment]
Does anyone have a workaround to import and build TensorFlow from a Bzlmod project ? I created a StackOverflow for this matter but haven't go any response yet : https://stackoverflow.com/questions/78413217/import-tensorflow-as-bazel-dependency-for-c-project

**Comment by fuhailin** [comment]
Is there any update?👀 
This is an important feature, I am willing to contribute my effort if possible

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.

**Comment by bc-lee** [comment]
> Hi,
> 
> Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.
> 
> The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.
> 
> Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.

This issue is about Bzlmod support, specifically adding a MODULE.bazel file for Bazel integration. The earlier comment seems to be a generic template about runtime bugs/hardware compatibility, which isn’t relevant to this request. To move this forward, it would be most helpful if someone familiar with Bazel/Bzlmod could take a look.

**Comment by mihaimaruseac** [comment]
TF depends on some external projects by commit hash instead of a released version. This makes it harder to bzlmod-ify it, sadly.

**Comment by wffurr** [comment]
You can use the `git_repository` rule to depend on a commit hash.  Here's an example from https://github.com/google/ink/blob/6bca16675fa6df7e31de6054f1620c11d82d3dc5/MODULE.bazel#L61

```
git_repository = use_repo_rule("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")

git_repository(
    name = "ink_stroke_modeler",
    commit = "748f84a30335b773e073fd1cf31c591be0a6b77e",  # Head as of 2025-06-17
    remote = "https://github.com/google/ink-stroke-modeler.git",
)
```

Happy to consult on bzlmod setup if you have a pending CL I can look at, ping me at wfurr@.

**Comment by mihaimaruseac** [comment]
Yeah, that's a possibility, but TF has complicated setup, afaik. Will reach out
